### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by yannikschaelte**

### CODE REVIEW

File: pyabc/population.py [hunk 1]
```diff
@@ -196,8 +196,8 @@ def get_weighted_distances(self) -> pandas.DataFrame:
         Returns
         -------
 
-        weighted_distances: pandas.DataFrame:
-            A pandas.DataFrame containing in column 'distance' the distances
+        weighted_distances: pd.DataFrame:
+            A pd.DataFrame containing in column 'distance' the distances
```
Comment (elbaraim) [inline pyabc/population.py]: I would write "A pandas DataFrame ... "
The abbreviation of `pd` is widely used, but it is not the exact name of the package

File: pyabc/storage/history.py [hunk 2]
```diff
@@ -123,64 +143,64 @@ def all_runs(self):
     @with_session
     def _pre_calculate_id(self):
         abcs = self._session.query(ABCSMC).all()
-        if len(abcs) == 1:
-            return abcs[0].id
+        if len(abcs) > 0:
```
Comment (elbaraim) [inline pyabc/storage/history.py]: why this change?

File: pyabc/storage/history.py [hunk 3]
```diff
@@ -123,64 +143,64 @@ def all_runs(self):
...
         -------
 
         df, w: pandas.DataFrame, np.ndarray
-
-        df:
-            is a DataFrame of parameters
-        w:
-            are the weights associated with each parameter
+            * df: a DataFrame of parameters
+            * w:  are the weights associated with each parameter
```
Comment (elbaraim) [inline pyabc/storage/history.py]: two empty spaces here ;)

File: pyabc/storage/history.py [hunk 4]
```diff
@@ -758,21 +816,26 @@ def get_population_strategy(self):
...
         Parameters
         ----------
-        m: int, optional
-            The model to query.
-            If omitted, all models are returned
-        t: str, optional
-            Can be "last" or "all"
+
+        m: int or None, optional (default = None)
+            The model to query. If omitted, all models are returned
```
Comment (elbaraim) [inline pyabc/storage/history.py]: full spot at the end of the sentence :dancer:

File: pyabc/storage/history.py [hunk 5]
```diff
@@ -758,21 +816,26 @@ def get_population_strategy(self):
...
+
+        t: int or str, optional (default = "last")
+            Can be "last" or "all", or a population index.
             In case of "all", all populations are returned.
-            If "last", only the last population is returned.
+            If "last", only the last population is returned, for an int value
+            only the corresponding population at that time index.
+
```
Comment (elbaraim) [inline pyabc/storage/history.py]: too picky: but the case of `population index` given is not explained :dancing_women:

File: pyabc/storage/history.py (hunk 2 above)
Comment (yannikschaelte) [reply pyabc/storage/history.py]: what stood there beforehand means that only if there is just 1 analysis in the db, then the id is set to it.
now the id is always set to the lastest added analysis, cause that's what I would expect (it is the most relevant one).

**Comment by yannikschaelte** [comment]
sorry, i had closed th pr already. these things are indeed true, I will add them in #56

File: pyabc/population.py (hunk 1 above)
Comment (yannikschaelte) [reply pyabc/population.py]: good point. I used pd here now cause I had changed the import to ``import pandas as pd``. Best would be to generate links to the respective classes here ...
